openquake/hazardlib/gsim/drouet_Alpes_2015.py

```diff
@@ -0,0 +1,681 @@
+    COEFFS = CoeffsTable(sa_damping=5, table="""\
+    IMT      c1       c2        c3        c4        c5       c6       c7        sigma    tau                                                                               
+	pgv 1.071377 -0.104440 -0.256368 -2.082968 0.175980 6.092497 -0.003354 0.596648 0.435828
+	pga 3.301210 -0.737892 -0.306729 -2.161341 0.160068 6.456162 -0.005434 0.681749 0.534690
+	0.010000 3.316019 -0.739661 -0.306739 -2.166328 0.160287 6.466172 -0.005424 0.684600 0.534966
+	0.020000 3.397746 -0.767832 -0.307173 -2.223926 0.165005 6.430227 -0.005301 0.718285 0.538337
+	0.030000 3.713900 -0.802179 -0.307387 -2.335400 0.170016 6.776541 -0.005022 0.765576 0.541424
+	0.040000 3.856821 -0.837170 -0.303832 -2.424559 0.178855 6.744506 -0.005145 0.794719 0.549611
+	0.050000 4.202453 -0.826429 -0.304496 -2.421348 0.171621 7.223255 -0.005539 0.806723 0.549383
+	0.075000 4.272509 -0.761342 -0.302493 -2.221265 0.153865 7.096516 -0.007126 0.770664 0.548284
+	0.100000 4.196424 -0.734942 -0.300902 -2.095881 0.145491 6.730797 -0.007958 0.739038 0.550258
+	0.150000 4.092752 -0.607407 -0.304960 -1.726689 0.111430 6.044787 -0.008780 0.684663 0.541002
+	0.200000 3.902236 -0.541433 -0.304370 -1.631331 0.107681 5.565939 -0.007979 0.649140 0.530486
+	0.250000 3.735820 -0.564165 -0.314904 -1.613639 0.108247 5.345891 -0.007363 0.626928 0.523275
+	0.300000 3.813539 -0.567288 -0.330693 -1.562863 0.096358 5.894730 -0.006721 0.610974 0.511050
+	0.400000 3.738093 -0.616395 -0.353019 -1.596618 0.097112 6.244053 -0.005635 0.596194 0.495412
+	0.500000 3.500535 -0.584768 -0.365189 -1.520070 0.093403 5.744979 -0.005183 0.584400 0.483265
+	0.750000 3.234558 -0.548882 -0.381135 -1.607358 0.102542 6.049448 -0.003642 0.576508 0.453998
+	1.000000 3.175584 -0.443241 -0.381813 -1.687741 0.109819 6.648249 -0.002547 0.573153 0.427804
+	1.250000 3.063413 -0.347957 -0.383817 -1.663268 0.106795 6.801522 -0.002023 0.571455 0.418229
+	1.500000 2.881863 -0.260619 -0.381584 -1.658755 0.110979 6.353724 -0.001851 0.570031 0.411255
+	1.750000 2.881210 -0.232600 -0.381592 -1.723555 0.115676 6.785918 -0.001748 0.566234 0.405577
+	2.000000 2.671433 -0.163676 -0.377968 -1.669152 0.112208 6.492545 -0.002183 0.563816 0.402956
+	3.000000 2.275590 0.247272 -0.326835 -1.646368 0.119810 6.194922 -0.001987 0.566383 0.384096
+    """)
+
+class DrouetAlpes2015Rrup(DrouetAlpes2015Rjb):
+    """
+    Implements GMPE developed by Douet & Cotton (2015) BSSA doi: 10.1785/0120140240.
+    Valid for vs30=800 m/s
+    """
+    
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        assert all(stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+                   for stddev_type in stddev_types)
+
+        C = self.COEFFS[imt]
+        mean = self._compute_mean(C, rup.mag, dists.rrup)
```

Comment:
If you pass the dists object rather than its attribute dists.rjb or dists.rrup then you would only need to modify the `_compute_mean` function to switch from rjb to rrup to rhypo.

Reply:
I don't really understand how this will help, I will have to pass another argument to the _compute_mean function to condition which distance to use, no?

Reply:
Looking again, the idea behind the comment was that if you passed `dists` rather than `dists.rjb` into the call to `_compute_mean` function in the DrouetAlpes2015Rjb, and also passed the `dists` object to the `_compute_term2` function and only inside the `_compute_term2` function indicate:
```python
def _compute_term2(self, C, mag, dists):
    """
    This computes the term f2 equation 8 Drouet & Cotton (2015)
    """
    return (C['c4'] + C['c5'] * mag) * np.log(np.sqrt(dists.rjb **2 + C['c6'] **2)) + C['c7'] * dists.rjb
```
Then in the subclasses you would only need to modify the `_compute_term2` function (and the `REQUIRES_DISTANCES`) in the subclasses and not need to repeat so much of the code of the `get_mean_and_stddevs` method. 
That said however, this is a cosmetic change to reduce the number of lines of code and doesn't really have any impact, so it is not critical.
Note, though, that changing the `REQUIRES_DISTANCES` attribute to reflect the distance metric used in each subclass is critical!

Reply:
I would still have to modify:
stddevs = self._get_stddevs(C, stddev_types, rup.mag,
                                    dists.rjb.shape[0])
in the get_mean_and_stddevs method. 
The Requires_distance attribute is now defined in each subclass, thanks.